### PR TITLE
Add mcore spec for full TE TransformerLayer

### PR DESCRIPTION
**Pull request by jbaczek**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by ShriyaPalsamudram** [comment]
jenkins

File: nemo/collections/nlp/models/language_modeling/megatron/gpt_full_te_layer_autocast_spec.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+            params_dtype=config.params_dtype,
+            get_rng_state_tracker=tensor_parallel.random.get_cuda_rng_tracker,
+            fuse_wgrad_accumulation=config.gradient_accumulation_fusion,
+            seq_length=None,  # used for jit warmup
+            micro_batch_size=None,  # used for jit warmup
+            sequence_parallel=config.sequence_parallel,
+            apply_residual_connection_post_layernorm=config.apply_residual_connection_post_layernorm,
+            autocast_dtype=precision,
+            #use_emha=False, # Use default 'False'
+            ub_tp_comm_overlap=config.tp_comm_overlap, # TODO: ub_tp_comm_overlap?
```
Comment (minitu) [inline nemo/collections/nlp/models/language_modeling/megatron/gpt_full_te_layer_autocast_spec.py:173]:
```suggestion
            ub_tp_comm_overlap=config.tp_comm_overlap,
```

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 2]
```diff
@@ -225,6 +227,8 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
 
         self.mcore_gpt = cfg.get('mcore_gpt', False)
         self.spec_name = cfg.get('name', '')
+        if cfg.get('fp8', False):
```
Comment (minitu) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:230]: I believe this is an artifact from Jimmy's memory fixes (the first two commits), which are now in main.
So those changes should go away as well.

**Comment by jbaczek** [comment]
Closing this one in favour of https://github.com/NVIDIA/NeMo/pull/8328

**Pull request closed without merging**